packages/dev/core/src/Misc/textureTools.ts

```diff
@@ -230,6 +231,179 @@ export function FromHalfFloat(value: number): number {
     return (s ? -1 : 1) * Math.pow(2, e - 15) * (1 + f / Math.pow(2, 10));
 }
 
+const ProcessAsync = async (
+    texture: BaseTexture,
+    width: number,
+    height: number,
+    face: number,
+    channels: { R: boolean; G: boolean; B: boolean; A: boolean },
+    lod: number,
+    resolve: (result: Uint8Array) => void,
+    reject: () => void
+): Promise<void> => {
+    const scene = texture.getScene()!;
+    const engine = scene.getEngine();
+
+    let lodPostProcess: PostProcess;
+
+    if (!texture.isCube) {
+        lodPostProcess = new PostProcess("lod", "lod", ["lod", "gamma"], null, 1.0, null, Texture.NEAREST_NEAREST_MIPNEAREST, engine);
+    } else {
+        const faceDefines = ["#define POSITIVEX", "#define NEGATIVEX", "#define POSITIVEY", "#define NEGATIVEY", "#define POSITIVEZ", "#define NEGATIVEZ"];
+        lodPostProcess = new PostProcess("lodCube", "lodCube", ["lod", "gamma"], null, 1.0, null, Texture.NEAREST_NEAREST_MIPNEAREST, engine, false, faceDefines[face]);
+    }
+
+    if (!lodPostProcess.getEffect().isReady()) {
+        // Try again later
+        lodPostProcess.dispose();
+
+        setTimeout(() => {
+            ProcessAsync(texture, width, height, face, channels, lod, resolve, reject);
+        }, 250);

```

Comment:
I know this was in the original code, so this is not a comment regarding this specific PR. Just a future TODO - if feels a bit off using the arbitrary value "250" . We should probably find a better way to do that.

Reply:
Can you replace with effect.onCompileObservable.addOnce... instead ???

Reply:
This is also a dangling promise.

Reply:
Oh, I didn't catch @sebavan's suggestion-- let me try that. But I restructured the code to avoid the weird promise logic.